### PR TITLE
#4837 - Only show cases in the users jurisdiction by default

### LINKED ISSUE #4837: Only show entities in the users jurisdiction by default on the mobile app [3]
### Problem Description
On the web app, there are filters like: Include cases from other jurisdiction. This is missing on the mobile app, causing much data that the user do not need being presented to the user.
### Possible Alternatives
None
### Development Specifications
* Don't show cases/contacts in the mobile app by default that are not inside the user's jurisdiction (same logic as in the web app https://github.com/hzi-braunschweig/SORMAS-Project/issues/2137); only do this for the main lists (not for case contacts)
* Add checkbox filters to the case and contact lists ("Include cases/contacts from other jurisdictions") in the mobile app that work similar to those in the web app and that will disable the restriction mentioned above (i.e. checking these filters will include entries from other jurisdictions); default unchecked
* It will be necessary to create a new `FilterCheckboxField` component for this; please try to style it in a way that fits the other filter components

### PR DESCRIPTION
**Pull request by carina29**
Fixes #4837

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12720

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12720

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12720

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12720

File: sormas-app/app/src/main/java/de/symeda/sormas/app/backend/caze/CaseDao.java [hunk 1]
```diff
@@ -754,54 +754,139 @@ public List<Case> queryByCriteria(CaseCriteria criteria, long offset, long limit
...
+			if (caseCriteria.getIncludeCasesFromOtherJurisdictions().equals(true)) {
+				whereUserFilterStatements.add(
+					where.or(
+						where.eq(Case.REPORTING_USER, currentUser.getId()),
+						where.eq(Case.SURVEILLANCE_OFFICER, currentUser.getId()),
+						where.eq(Case.CASE_OFFICER, currentUser.getId())));
+			}
+
+			switch (jurisdictionLevel) {
+			case REGION:
```
Comment (MateStrysewske) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/backend/caze/CaseDao.java:793]: Nation and region level are not necessary; users with these jurisdiction levels don't have access to the mobile app

File: sormas-app/app/src/main/java/de/symeda/sormas/app/backend/contact/ContactDao.java [hunk 2]
```diff
@@ -191,53 +195,199 @@ public List<Contact> queryByCriteria(ContactCriteria criteria, long offset, long
...
+			if (contactCriteria.getIncludeContactsFromOtherJurisdictions().equals(true)) {
+				whereJurisdictionFilterStatements.add(
+					where.or(
+						where.raw(Case.TABLE_NAME + "." + Case.REPORTING_USER + "= '" + currentUser.getId() + "'"),
+						where.raw(Case.TABLE_NAME + "." + Case.SURVEILLANCE_OFFICER + "= '" + currentUser.getId() + "'"),
+						where.raw(Case.TABLE_NAME + "." + Case.CASE_OFFICER + "= '" + currentUser.getId() + "'")));
+			}
+
+			switch (jurisdictionLevel) {
+			case REGION:
```
Comment (MateStrysewske) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/backend/contact/ContactDao.java:248]: Same here

File: sormas-app/app/src/main/java/de/symeda/sormas/app/backend/contact/ContactDao.java [hunk 3]
```diff
@@ -191,53 +195,199 @@ public List<Contact> queryByCriteria(ContactCriteria criteria, long offset, long
...
+		final JurisdictionLevel jurisdictionLevel = currentUser.getJurisdictionLevel();
+
+		// whoever created it or is assigned to it is allowed to access it
+		if (contactCriteria == null || contactCriteria.getIncludeContactsFromOtherJurisdictions()) {
+			whereUserFilterStatements
+				.add(where.or(where.eq(Contact.REPORTING_USER, currentUser.getId()), where.eq(Contact.CONTACT_OFFICER, currentUser.getId())));
+		}
+
+		switch (jurisdictionLevel) {
+		case REGION:
```
Comment (MateStrysewske) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/backend/contact/ContactDao.java:319]: Region is unnecessary here as well

File: sormas-app/app/src/main/java/de/symeda/sormas/app/backend/contact/ContactDao.java [hunk 4]
```diff
@@ -191,53 +195,199 @@ public List<Contact> queryByCriteria(ContactCriteria criteria, long offset, long
...
+			}
+		}
+
+		if (!whereJurisdictionFilterStatements.isEmpty()) {
+			where.or(whereJurisdictionFilterStatements.size());
+		}
+		return where;
+	}
+
+	public Where<Contact, Long> createJurisdictionFilterWithoutCase(Where<Contact, Long> where, ContactCriteria contactCriteria) throws SQLException {
```
Comment (MateStrysewske) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/backend/contact/ContactDao.java:302]: Call this `createJurisdictionFilter` and the other method `createJurisdictionFilterForCase`

**Review by MateStrysewske — CHANGES_REQUESTED** [review]
The code contains logic that is not necessary in the mobile app. There's no need for ANY additionally executed code when `includeCasesFromOtherJurisdictions` or `includeContactsFromOtherJurisdictions` is true because, by default, the lists show all the entities in the database anyway.

The only thing that's necessary is that, if the new filter is false, IN ADDITION to the existing code, add predicates that make sure that only data from the same jurisdiction as the users' is shown (i.e. the jurisdiction switch part).

That means:
- Remove the new code that checks the reporting user, responsible user, etc.
- ONLY call the whole jurisdiction check logic when the new filter is set to false

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12720

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12720

**Comment by carina29** [comment]
> The code contains logic that is not necessary in the mobile app. There's no need for ANY additionally executed code when `includeCasesFromOtherJurisdictions` or `includeContactsFromOtherJurisdictions` is true because, by default, the lists show all the entities in the database anyway.
> 
> The only thing that's necessary is that, if the new filter is false, IN ADDITION to the existing code, add predicates that make sure that only data from the same jurisdiction as the users' is shown (i.e. the jurisdiction switch part).
> 
> That means:
> 
> * Remove the new code that checks the reporting user, responsible user, etc.
> * ONLY call the whole jurisdiction check logic when the new filter is set to false

I've removed the parts you've mentioned.

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12720

File: sormas-app/app/src/main/java/de/symeda/sormas/app/backend/caze/CaseDao.java [hunk 5]
```diff
@@ -779,64 +779,49 @@ public Where createJurisdictionFilter(Where<Case, Long> where, CaseCriteria case
...
-		if (jurisdictionLevel != JurisdictionLevel.NATION) {
-			//whoever created the case or is assigned to it is allowed to access it
-			if (caseCriteria.getIncludeCasesFromOtherJurisdictions().equals(true)) {
-				whereUserFilterStatements.add(
-					where.or(
-						where.eq(Case.REPORTING_USER, currentUser.getId()),
-						where.eq(Case.SURVEILLANCE_OFFICER, currentUser.getId()),
-						where.eq(Case.CASE_OFFICER, currentUser.getId())));
-			}
+		//whoever created the case or is assigned to it is allowed to access it
```
Comment (MateStrysewske) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/backend/caze/CaseDao.java:782]: Unnecessary comment

**Review by MateStrysewske — CHANGES_REQUESTED** [review]
I'll do some testing after the release :)

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12720

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12720

**Review by MateStrysewske — CHANGES_REQUESTED** [review]
Please change the text colour of the filter; should be the same colour as e.g. the text input of the free text filter (not the placeholder); also don't make it bold (I believe it is right now). Logic seems to work fine :)

**Comment by carina29** [comment]
> Please change the text colour of the filter; should be the same colour as e.g. the text input of the free text filter (not the placeholder); also don't make it bold (I believe it is right now). Logic seems to work fine :)

I've changed the font and colour as you've suggested ( is the same styling as we have for the filter text):
image

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12720

**Review by MateStrysewske — APPROVED** [review]

**Pull request merged**